### PR TITLE
[Tig-797] signup hooks

### PR DESCRIPTION
**Pull request by AveryDodd**
## Ticket(s)

* [[Tig-797]Update/Add hooks to use new attributes](https://app.clickup.com/t/9009201449/TIG-797)

## Description
Sorry, lots of file changes but the new hooks are almost identical in structure. 

- Added signup hooks to interact with new attributes.
- Updated existing signup hooks to use new attributes
- Added tests for new hooks 
- Updated mock data 
- Removed superfluous user-preference hooks.  

## To do

- [ ] Added hooked data to sign-up components

### CODE REVIEW

File: src/hooks/use-signup-ignore-actions.hook.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export const useSignupIgnoreActions = () => {
+  const { userProfile } = useGetCurrentUserProfile();
+  const { updateUserProfile } = useUpdateCurrentUserProfile();
+  const [ignoreActions, setIgnoreActions] = React.useState(defaultIgnoreActions);
+  
+  useEffect(() => {
+    if (
+      userProfile !== undefined
+    ) {
+      setIgnoreActions(userProfile['custom:ignoreActions'].split(','));
```
Comment (stephenbelyea) [inline src/hooks/use-signup-ignore-actions.hook.js:17]: Does the array of `ignoreActions` come as a string from the API?

File: src/hooks/use-get-current-user-profile.test.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ describe('useGetCurrentUserProfile', () => {
     let result;
     const mockUserWithAvatar = {
       ...mockSimpleUser,
-      avatar: avatarOptions[0],
+      avatar: avatarOptions[0].id,
```
Comment (stephenbelyea) [inline src/hooks/use-get-current-user-profile.test.js:40]: Initially, I was thinking we could include the full `avatar` object with the UI-facing user profile. Like match it by ID when we initially receive the API response. 

Not something to worry about right away, as we can probably just have a separate `useGetUserAvatar` hook or something that matches the stored avatar object for when we need to show it in the profile image or other areas around the app.

File: src/hooks/use-signup-ignore-actions.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    expect(result.current.setIgnoreActions).toBeDefined();
+    expect(result.current.sendIgnoreActionsRequest).toBeDefined();
+  });
+
+  it('returns updated values when setIgnoreActions is called', async () => {
+    const { result } = renderHook(useSignupIgnoreActions);
+
+    expect(result.current.ignoreActions).toEqual(defaultIgnoreActions);
+
+    act(() => result.current.setIgnoreActions(updatedIgnoreActions));
```
Comment (stephenbelyea) [inline src/hooks/use-signup-ignore-actions.test.js:24]: Did you have any issues with mocking and those annoying "you used the wrong `act`" warnings in the terminal while setting up these tests?

File: src/hooks/use-signup-info-secondary.hook.js [hunk 4]
```diff
@@ -18,14 +19,37 @@ export const ageRanges = [
...
   const { updateUserProfile } = useUpdateCurrentUserProfile();
   const [values, setValues] = React.useState(defaultValues);
 
   const provinces = getProvincesForCountry(values.country);
 
   // Temporary stand-in for API request method
-  const sendValuesRequest = async () => await updateUserProfile(values);
+  const sendValuesRequest = async () => await updateUserProfile(
+    {
+      'preferred_username': values.nickname,
```
Comment (stephenbelyea) [inline src/hooks/use-signup-info-secondary.hook.js:32]: Could be worth noting here that we'll want to sanitize and validate user entry eventually before submitting to the API - but this is great for now as it gets us up and running!

File: src/__mocks__/user-attributes.mock.js [hunk 5]
```diff
@@ -12,14 +12,14 @@ export const mockSimpleUser = {
   username: 'userperson',
   isAdmin: false,
   type: USER_TYPE.MEMBER,
-  ageRange: '1-5',
+  ageRange: '1-6',
```
Comment (stephenbelyea) [inline src/__mocks__/user-attributes.mock.js:15]: Just a note that we do have an array of `ageRange` objects stored in this file: `src/hooks/use-signup-info-secondary.hook.js`.

Probably worth breaking them out as a dedicated `ageRangeOptions` constant, but it's there for now.

**Review by stephenbelyea — APPROVED** [review]
These all look great! Couple of small questions and notes. 

How are you testing these? Did you manually create a user with the new fields so you can run through?

File: src/hooks/use-signup-ignore-actions.hook.js (hunk 1 above)
Comment (AveryDodd) [reply src/hooks/use-signup-ignore-actions.hook.js:17]: It does indeed. The only options for types it supports are string or number unfortunately

File: src/hooks/use-get-current-user-profile.test.js (hunk 2 above)
Comment (AveryDodd) [reply src/hooks/use-get-current-user-profile.test.js:40]: Yeah no problem. This is something I had debated about and decided to just picked a direction but it wouldn't be hard to switch or add a new get hook!

File: src/hooks/use-signup-ignore-actions.test.js (hunk 3 above)
Comment (AveryDodd) [reply src/hooks/use-signup-ignore-actions.test.js:24]: I don't believe so, though I'm not the most verbose tester so I may have missed something

File: src/__mocks__/user-attributes.mock.js (hunk 5 above)
Comment (AveryDodd) [reply src/__mocks__/user-attributes.mock.js:15]: Good to know!

**Pull request merged**